src/pybind/mgr/rbd_support/module.py

```diff
@@ -483,13 +499,687 @@ def get_perf_counters(self, pool_spec, sort_by):
+            elif prefix == 'remove':
+                return self.queue_remove(cmd['image_spec'])
+            elif prefix == 'trash remove':
+                return self.queue_trash_remove(cmd['image_id_spec'])
+            elif prefix == 'migration execute':
+                return self.queue_migration_execute(cmd['image_spec'])
+            elif prefix == 'migration commit':
+                return self.queue_migration_commit(cmd['image_spec'])
+            elif prefix == 'migration abort':
+                return self.queue_migration_abort(cmd['image_spec'])
+            elif prefix == 'cancel':
+                return self.task_cancel(cmd['task_id'])
+            elif prefix == 'list':
+                return self.task_list(cmd.get('task_id'))
+
+        raise NotImplementedError(cmd['prefix'])
+
+
+class Module(MgrModule):
+    COMMANDS = [
+        {
+            "cmd": "rbd perf image stats "
+                   "name=pool_spec,type=CephString,req=false "
+                   "name=sort_by,type=CephChoices,strings="
+                   "write_ops|write_bytes|write_latency|"
+                   "read_ops|read_bytes|read_latency,"
+                   "req=false ",
+            "desc": "Retrieve current RBD IO performance stats",
+            "perm": "r"
+        },
+        {
+            "cmd": "rbd perf image counters "
+                   "name=pool_spec,type=CephString,req=false "
+                   "name=sort_by,type=CephChoices,strings="
+                   "write_ops|write_bytes|write_latency|"
+                   "read_ops|read_bytes|read_latency,"
+                   "req=false ",
+            "desc": "Retrieve current RBD IO performance counters",
+            "perm": "r"
+        },
+        {
+            "cmd": "rbd task flatten "
+                   "name=image_spec,type=CephString",
+            "desc": "Flatten a cloned image asynchronously in the background",
+            "perm": "w"
+        },
+        {
+            "cmd": "rbd task remove "
+                   "name=image_spec,type=CephString",
+            "desc": "Remove an image asynchronously in the background",
+            "perm": "w"
+        },
+        {
+            "cmd": "rbd task trash remove "
+                   "name=image_id_spec,type=CephString",
+            "desc": "Remove an image from the trash asynchronously in the background",
+            "perm": "w"
+        },
+        {
+            "cmd": "rbd task migration execute "
+                   "name=image_spec,type=CephString",
+            "desc": "Execute an image migration asynchronously in the background",
+            "perm": "w"
+        },
+        {
+            "cmd": "rbd task migration commit "
+                   "name=image_spec,type=CephString",
+            "desc": "Commit an executed migration asynchronously in the background",
+            "perm": "w"
+        },
+        {
+            "cmd": "rbd task migration abort "
```

Comment:
Just thinking, may be it is better to use `rbd task add <command>`  (i.e. `rbd task add migration abort ...` instead of `rbd task migration abort`) so it would be easier to distinguish between task add commands and other task maintenance commands like `rbd task list` or `rbd task cancel`?

Reply:
To be honest, that's how I had originally implemented it but then I yanked it. Since it sounds like it makes sense to you as well, I will re-add `add`.